### PR TITLE
Feature 2040 user defined labels in config

### LINKED ISSUE #2040: User defined labels in scan configuration which are available in report
# Situation

At he moment the only information which is contained in the report is the job uuid. The job uuid is unique, but generated by SecHub and very generic.

# Wanted

As I user, I want to be able to add additional information to the scan configuration which end up in the report as key value pairs.

Take for example this configuration:

~~~json
{
  "apiVersion" : "1.0",
  "data" : {
    "binaries" : [ {
      "name" : "myproject",
      "fileSystem" : {
        "folders" : [ "myproject/code" ]
      }
    } ]
  },
  "codeScan" : {
    "use" : ["myproject"]
  }
}
~~~

The report will get the a job UUID and that will be in the report as well. For example: 60658942-bda8-11ed-9405-23831bdc0234. 

~~~json
{
   "result": {
      "count": 15,
      "findings": [
         {
            "id": 1,
            "description": "Unexpected property leak",
            "name": "Unexpected property leak.",
            "severity": "MEDIUM",
            "code": {
               "location": "com/myproject/code/controller/AuthorController.kt",
               "line": 23,
               "column": 0
            },
            "type": "codeScan"
         },
         …
         {
            "id": 15,
            "description": "MD2, MD4 and MD5 are weak hash functions",
            "name": "MD2, MD4 and MD5 are weak hash functions.",
            "severity": "MEDIUM",
            "code": {
               "location": "com/myproject/code/controller/UserController.kt",
               "line": 61,
               "column": 0
            },
            "type": "codeScan"
         }
      ]
   },
   "reportVersion": "1.0",
   "messages": [],
   "trafficLight": "YELLOW",
   "jobUUID": "60658942-bda8-11ed-9405-23831bdc0234",
   "status": "SUCCESS"
}
~~~

Now let's assume the users want to provide additional information (context) for the scan. To add them the users add labels:

~~~json
{
  "apiVersion" : "1.0",
  "metaData": {
     "labels": {
        "stage": "testing", 
        "purpose": "quality assurance" 
    }
  },
  "data" : {
    "binaries" : [ {
      "name" : "myproject",
      "fileSystem" : {
        "folders" : [ "myproject/code" ]
      }
    } ]
  },
  "codeScan" : {
    "use" : ["myproject"]
  }
}
~~~

The labels will be in the report:

~~~json
{
   "result": {
      "count": 15,
      "findings": [
         {
            "id": 1,
            "description": "Unexpected property leak",
            "name": "Unexpected property leak.",
            "severity": "MEDIUM",
            "code": {
               "location": "com/myproject/code/controller/AuthorController.kt",
               "line": 23,
               "column": 0
            },
            "type": "codeScan"
         },
         …
         {
            "id": 15,
            "description": "MD2, MD4 and MD5 are weak hash functions",
            "name": "MD2, MD4 and MD5 are weak hash functions.",
            "severity": "MEDIUM",
            "code": {
               "location": "com/myproject/code/controller/UserController.kt",
               "line": 61,
               "column": 0
            },
            "type": "codeScan"
         }
      ]
   },
   "reportVersion": "1.0",
   "messages": [],
  "metaData": {
     "labels": {
        "stage": "testing", 
        "purpose": "quality assurance" 
    }
  }
   "trafficLight": "YELLOW",
   "jobUUID": "60658942-bda8-11ed-9405-23831bdc0234",
   "status": "SUCCESS"
}
~~~

Now it is clear for the user, this report was created on the testing stage for quality assurance purposes. This is how adding labels can help the users to identify reports easier with context information which matter to them.

### LINKED ISSUE #2125: Tests inside `ThymeLeafHTMLReportingTest` are green but ouput is empty
## Situation
While developing #2040 and implementing the HTML parts I wanted to test the output and extend tests for it.
`ThymeLeafHTMLReportingTest` was created to test HTML output without starting SecHub server.

But... The output is currently always an empty test - and the tests are still green!

## Wanted
- [x] Fix the problem of the empty html output inside the test results
- [x] Change the tests - add  checks that at least content contains some of the expected vulnerabilities. It shall fail for empty reports..

### LINKED ISSUE #2129: SimpleStringUtils.hasOnlyAlphabeticDigitOrAdditionalAllowedCharacters wrong behavior
## Situation
In class `SimpleStringUtils` the method `hasOnlyAlphabeticDigitOrAdditionalAllowedCharacters`
does have following javadoc:
```javadoc
/**
     * Will test if given string does contain only alphabetic characters (a-z,A-Z),
     * digits (0-9) or additionally allowed characters.
     *
     * @param string            the content to inspect.
     * @param additionalAllowed
     * @return <code>true</code> when only allowed characters are contained, or
     *         given string is <code>null</code> or empty. <code>false</code>
     *         otherwise.
     */
```

But currently a string containing german umlauts is accepted because `Character.isAlphabetic()` is used
to check!

## Wanted
The method shall behave like defined in javadoc

### PR DESCRIPTION
**Pull request by de-jcup**
This PR
- closes #2040
- closes #2125
- closes #2129

### CODE REVIEW

**Review by Jeeppler — CHANGES_REQUESTED** [review]
Overall looks good. However, I found a number of small issues. Please fix those.

In addition, please verify whether your pull-request fixes this issue as well: https://github.com/mercedes-benz/sechub/issues/2071. If it does, please add it to the description.

File: sechub-commons-model/src/test/java/com/mercedesbenz/sechub/commons/model/SecHubConfigurationModelValidatorTest.java [hunk 1]
```diff
@@ -735,6 +736,266 @@ void secret_scan__config_with_data() throws Exception {
...
+        /* execute */
+        SecHubConfigurationModelValidationResult result = validatorToTest.validate(model);
+
+        /* test */
+        assertHasError(result, SecHubConfigurationModelValidationError.METADATA_LABEL_VALUE_TOO_BIG);
+        assertEquals(1, result.getErrors().size());
+    }
+
+    @ParameterizedTest
+    @ValueSource(strings = { "$variable1", "!something", "<html>", "label:with:colon" })
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-commons-model/src/test/java/com/mercedesbenz/sechub/commons/model/SecHubConfigurationModelValidatorTest.java:913]: While adding your suggestions, I ound here a problem in SimpleStringUtils implementation when using umlauts.
Created another issue: 
- #2129

[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-commons-model/src/test/java/com/mercedesbenz/sechub/commons/model/SecHubConfigurationModelValidatorTest.java:913]: Will fix this inside the PR as well.

File: sechub-scan/src/main/resources/templates/report/html/scanresult.html [hunk 2]
```diff
@@ -60,6 +67,18 @@
 					<td>Scan result Report for Job:<span th:text="${jobuuid}">job-uuid</span>
 					</td>
 				</tr>
+				<tr class="reportlabels" th:if="${metaData.isPresent() && !metaData.get().getLabels().isEmpty()}" >
+                    <td></td>
+                    <td>
+                        <table class='reportMetaDataTable' >
+                           <tr th:each="entry : ${metaData.get().getLabels()}">
+                                <td class="metaDataLabelKey" th:text="${entry.key}">key</td>
+                                <td>:</td>
```
[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-scan/src/main/resources/templates/report/html/scanresult.html:76]: I need this to identify the parts inside the tests. So I will keep it as is.

[the remark this answers is not in the document]
Comment (de-jcup) [reply sechub-scan/src/main/resources/templates/report/html/scanresult.html:76]: Clarified - we are talking about : here and not about the class :-)

**Review by Jeeppler — APPROVED** [review]
Looks good 👍

**Pull request merged**